src/Util/JSON.php

```diff
@@ -50,4 +66,78 @@ public static function decode(string $data)
 
         return $decodedData;
     }
+
+    /**
+     * Performs sanity checks on data that shall be encoded to JSON.
+     *
+     * @param mixed $data     The data to sanitize
+     * @param int   $maxDepth The maximum depth to walk through `$data`
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    private static function sanitizeData($data, int $maxDepth)
+    {
+        if ($maxDepth < 0) {
+            throw new JsonException('Max depth limit of data to be sanitized reached.');
```

Comment:
I would not throw an exception here, but stop silently and truncate the serialization, like in our serializer.

Reply:
I'm not convinced that we should remove this exception: `json_encode` will throw an error anyway if the max depth limit (3rd argument, set to `512` by default) is reached. Plus, this check also ensures that the function is always called with a valid argument